src/hotspot/share/opto/superword.cpp

```diff
@@ -1691,42 +1743,52 @@ void SuperWord::filter_packs_for_alignment() {
   AlignmentSolution const* current = new TrivialAlignmentSolution();
   int mem_ops_count = 0;
   int mem_ops_rejected = 0;
+  int new_packset_length = 0;
   for (int i = 0; i < _packset.length(); i++) {
-    Node_List* p = _packset.at(i);
-    if (p != nullptr) {
-      if (p->at(0)->is_Load() || p->at(0)->is_Store()) {
-        mem_ops_count++;
-        // Find solution for pack p, and filter with current solution.
-        const AlignmentSolution* s = pack_alignment_solution(p);
-        const AlignmentSolution* intersect = current->filter(s);
+    Node_List* pack = _packset.at(i);
+    assert(pack != nullptr, "no nullptr in packset");
+    bool keep = true;
+    if (pack->at(0)->is_Load() || pack->at(0)->is_Store()) {
+      mem_ops_count++;
+      // Find solution for pack p, and filter with current solution.
+      const AlignmentSolution* s = pack_alignment_solution(pack);
+      const AlignmentSolution* intersect = current->filter(s);
 
 #ifndef PRODUCT
-        if (is_trace_align_vector()) {
-          tty->print("  solution for pack:         ");
-          s->print();
-          tty->print("  intersection with current: ");
-          intersect->print();
-        }
+      if (is_trace_align_vector()) {
+        tty->print("  solution for pack:         ");
+        s->print();
+        tty->print("  intersection with current: ");
+        intersect->print();
+      }
 #endif
 
-        if (intersect->is_empty()) {
-          // Solution failed or is not compatible, remove pack i.
+      if (intersect->is_empty()) {
+        // Solution failed or is not compatible, remove pack i.
 #ifndef PRODUCT
-          if (is_trace_superword_rejections() || is_trace_align_vector()) {
-            tty->print_cr("Rejected by AlignVector:");
-            p->at(0)->dump();
-          }
-#endif
-          _packset.at_put(i, nullptr);
```

Comment:
Note: I now shift the surviving (keep) packs down, rather than setting `nullptr` and `compress_packset` below.